### PR TITLE
PN-9861 Create "Awaiting Parts Report" for Work Orders After Quote Ap…

### PR DESCRIPTION
**Pull request by abhishekjir**
…proval

### CODE REVIEW

File: PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql [hunk 1]
```diff
@@ -0,0 +1,570 @@
...
+    @Level8 VARCHAR(MAX) = NULL,    
+    @Level9 VARCHAR(MAX) = NULL,    
+    @Level10 VARCHAR(MAX) = NULL,    
+    @IsDownload BIT = NULL   
+
+	DECLARE @Stage varchar(300) = NULL  
+    DECLARE @CustomerId varchar(100) = NULL 
+
+	DECLARE @ApprovedQuoteStatusId INT = 0
+	SELECT @ApprovedQuoteStatusId = WorkOrderQuoteStatusId FROM WorkOrderQuoteStatus WHERE Description = 'Approved';
```
Comment (amitghediyasilverxis) [inline PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql]: Add DBO WITH(nolock)

File: PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql [hunk 2]
```diff
@@ -0,0 +1,570 @@
...
+		INNER JOIN DBO.Stockline STK WITH (NOLOCK) ON tmpWOM.ItemMasterId = STK.ItemMasterId AND tmpWOM.ConditionId = STK.ConditionId AND STK.IsParent = 1    
+		 
+	WHERE    
+		WO.CustomerId=ISNULL(@CustomerId,WO.CustomerId)
+		AND ISNULL(WO.IsDeleted, 0) = 0  		 
+		AND ISNULL(WO.IsActive, 1) = 1 
+		AND  ISNULL(WO.WorkOrderStatusId, 0) != 2 -----WO Not Closed  
+		AND (ISNULL(@Stage,'') ='' OR WOS.WorkOrderStageId IN (SELECT value FROM String_split(ISNULL(@Stage,WOS.WorkOrderStageId), ',')))     
+		AND  ISNULL(WOPN.WorkOrderStatusId, 0) != 2 AND  ISNULL(WOPN.IsClosed, 0) != 1 -----MPN Not Closed  
+		--AND WO.WorkOrderId NOT IN (ISNULL(woshi.WorkOrderId,0))
```
Comment (amitghediyasilverxis) [inline PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql]: Remove unnecessary code.

File: PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql [hunk 3]
```diff
@@ -0,0 +1,570 @@
...
+
+	INSERT INTO #AwaitingPartsData
+	SELECT DISTINCT 0 AS TotalRecordsCount,    
+		WO.WorkOrderId,  
+		MAX(UPPER(WO.WorkOrderNum)) 'wonum',
+		MAX(UPPER(C.Name)) 'customername',
+		MAX(UPPER(WOQ.QuoteNumber)) 'woqnum',
+		MAX(UPPER(IM.partnumber)) 'pn',    
+		MAX(UPPER(IM.PartDescription)) 'pnDescription',    
+		--MAX(ISNULL((SELECT ISNULL(MAX(RepairOrderNumber),0) FROM dbo.RepairOrder rep WITH(NOLOCK) where rep.RepairOrderId = SL.RepairOrderId),0)) 'ronum',
```
Comment (amitghediyasilverxis) [inline PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql]: Remove unnecessary code.

File: PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql [hunk 4]
```diff
@@ -0,0 +1,570 @@
...
+		MAX(WO.MasterCompanyId) MasterCompanyId  
+
+	FROM DBO.WorkOrder WO WITH (NOLOCK)      
+		INNER JOIN DBO.WorkOrderWorkFlow WOWF WITH (NOLOCK) ON WOWF.WorkOrderId = WO.WorkOrderId     
+		INNER JOIN DBO.WorkOrderPartNumber WOPN WITH (NOLOCK) ON WOWF.WorkOrderPartNoId = WOPN.ID    
+		INNER JOIN dbo.WorkOrderManagementStructureDetails MSD WITH (NOLOCK) ON MSD.ModuleID = @ModuleID AND MSD.ReferenceID = WOPN.ID    
+		INNER JOIN DBO.WorkOrderQuote WOQ WITH (NOLOCK) ON WO.WorkOrderId = WOQ.WorkOrderId AND QuoteStatusId = @ApprovedQuoteStatusId
+		LEFT JOIN DBO.WorkOrderQuoteDetails WQD WITH (NOLOCK) ON WOQ.WorkOrderQuoteId = WQD.WorkOrderQuoteId  
+		LEFT JOIN DBO.Customer C WITH (NOLOCK) ON C.CustomerId = WO.CustomerId  
+		LEFT JOIN dbo.EntityStructureSetup ES ON ES.EntityStructureId=MSD.EntityMSID
```
Comment (amitghediyasilverxis) [inline PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql]: add nolock

File: PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql [hunk 5]
```diff
@@ -0,0 +1,570 @@
...
+		LEFT JOIN [dbo].TimeZone TZ WITH(NOLOCK) ON le.TimeZoneId = TZ.TimeZoneId
+	    LEFT JOIN #tmpMultipleWOMStocklineKit tmpWOM WITH (NOLOCK) ON tmpWOM.[WorkOrderId] = WO.[WorkOrderId]
+		LEFT JOIN DBO.ItemMaster AS IM WITH (NOLOCK) ON tmpWOM.ItemMasterId = IM.ItemMasterId    
+		LEFT JOIN DBO.WorkOrderMaterialsKit AS WOM WITH (NOLOCK) ON tmpWOM.WorkOrderMaterialsId = WOM.WorkOrderMaterialsKitId
+		LEFT JOIN DBO.Stockline STK WITH (NOLOCK) ON tmpWOM.ItemMasterId = STK.ItemMasterId AND tmpWOM.ConditionId = STK.ConditionId AND STK.IsParent = 1    
+		 
+	WHERE    
+		WO.CustomerId=ISNULL(@CustomerId,WO.CustomerId)
+		AND ISNULL(WO.IsDeleted, 0) = 0  
+		--CAST((select [dbo].[ConvertUTCtoLocal] (WO.OpenDate,TZ.Description)) AS DATE) BETWEEN CAST(@Fromdate AS DATE) AND CAST(@Todate AS DATE)  AND
```
Comment (amitghediyasilverxis) [inline PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql]: Remove unnecessary code.

File: PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql [hunk 6]
```diff
@@ -0,0 +1,570 @@
...
+	WHERE    
+		WO.CustomerId=ISNULL(@CustomerId,WO.CustomerId)
+		AND ISNULL(WO.IsDeleted, 0) = 0  
+		--CAST((select [dbo].[ConvertUTCtoLocal] (WO.OpenDate,TZ.Description)) AS DATE) BETWEEN CAST(@Fromdate AS DATE) AND CAST(@Todate AS DATE)  AND  
+		 
+		AND ISNULL(WO.IsActive, 1) = 1 
+		AND  ISNULL(WO.WorkOrderStatusId, 0) != 2 -----WO Not Closed  
+		AND (ISNULL(@Stage,'') ='' OR WOS.WorkOrderStageId IN (SELECT value FROM String_split(ISNULL(@Stage,WOS.WorkOrderStageId), ',')))     
+		AND  ISNULL(WOPN.WorkOrderStatusId, 0) != 2 AND  ISNULL(WOPN.IsClosed, 0) != 1 -----MPN Not Closed  
+		--AND WO.WorkOrderId NOT IN (ISNULL(woshi.WorkOrderId,0))
```
Comment (amitghediyasilverxis) [inline PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql]: Remove unnecessary code.

File: PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql [hunk 7]
```diff
@@ -0,0 +1,570 @@
...
+
+	FROM DBO.SubWorkOrder SWO WITH (NOLOCK)    
+		INNER JOIN DBO.WorkOrder WO WITH (NOLOCK) ON SWO.WorkOrderId = WO.WorkOrderId 
+		INNER JOIN DBO.WorkOrderWorkFlow WOWF WITH (NOLOCK) ON WOWF.WorkOrderId = WO.WorkOrderId     
+		INNER JOIN DBO.WorkOrderPartNumber WOPN WITH (NOLOCK) ON WOWF.WorkOrderPartNoId = WOPN.ID    
+		INNER JOIN dbo.WorkOrderManagementStructureDetails MSD WITH (NOLOCK) ON MSD.ModuleID = @ModuleID AND MSD.ReferenceID = WOPN.ID    
+		INNER JOIN DBO.WorkOrderQuote WOQ WITH (NOLOCK) ON WO.WorkOrderId = WOQ.WorkOrderId AND QuoteStatusId = @ApprovedQuoteStatusId
+		LEFT JOIN DBO.WorkOrderQuoteDetails WQD WITH (NOLOCK) ON WOQ.WorkOrderQuoteId = WQD.WorkOrderQuoteId  
+		LEFT JOIN DBO.Customer C WITH (NOLOCK) ON C.CustomerId = WO.CustomerId  
+		LEFT JOIN dbo.EntityStructureSetup ES ON ES.EntityStructureId=MSD.EntityMSID
```
Comment (amitghediyasilverxis) [inline PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql]: Add nolock

File: PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql [hunk 8]
```diff
@@ -0,0 +1,570 @@
...
+		INNER JOIN DBO.Stockline STK WITH (NOLOCK) ON tmpWOM.ItemMasterId = STK.ItemMasterId AND tmpWOM.ConditionId = STK.ConditionId AND STK.IsParent = 1    
+		 
+	WHERE    
+		WO.CustomerId=ISNULL(@CustomerId,WO.CustomerId)
+		AND ISNULL(WO.IsDeleted, 0) = 0  		 
+		AND ISNULL(WO.IsActive, 1) = 1 
+		AND  ISNULL(WO.WorkOrderStatusId, 0) != 2 -----WO Not Closed  
+		AND (ISNULL(@Stage,'') ='' OR WOS.WorkOrderStageId IN (SELECT value FROM String_split(ISNULL(@Stage,WOS.WorkOrderStageId), ',')))     
+		AND  ISNULL(WOPN.WorkOrderStatusId, 0) != 2 AND  ISNULL(WOPN.IsClosed, 0) != 1 -----MPN Not Closed  
+		--AND WO.WorkOrderId NOT IN (ISNULL(woshi.WorkOrderId,0))
```
Comment (amitghediyasilverxis) [inline PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql]: Remove unnecessary code.

File: PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql [hunk 9]
```diff
@@ -0,0 +1,570 @@
...
+
+	INSERT INTO #AwaitingPartsData
+	SELECT DISTINCT 0 AS TotalRecordsCount,    
+		SWO.SubWorkOrderId,  
+		MAX(UPPER(SWO.SubWorkOrderNo)) 'wonum',
+		MAX(UPPER(C.Name)) 'customername',
+		MAX(UPPER(WOQ.QuoteNumber)) 'woqnum',
+		MAX(UPPER(IM.partnumber)) 'pn',    
+		MAX(UPPER(IM.PartDescription)) 'pnDescription',    
+		--MAX(ISNULL((SELECT ISNULL(MAX(RepairOrderNumber),0) FROM dbo.RepairOrder rep WITH(NOLOCK) where rep.RepairOrderId = SL.RepairOrderId),0)) 'ronum',
```
Comment (amitghediyasilverxis) [inline PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql]: Remove unnecessary code.

File: PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql [hunk 10]
```diff
@@ -0,0 +1,570 @@
...
+
+	FROM DBO.SubWorkOrder SWO WITH (NOLOCK)
+		INNER JOIN DBO.WorkOrder WO WITH (NOLOCK) ON SWO.WorkOrderId = WO.WorkOrderId     
+		INNER JOIN DBO.WorkOrderWorkFlow WOWF WITH (NOLOCK) ON WOWF.WorkOrderId = WO.WorkOrderId     
+		INNER JOIN DBO.WorkOrderPartNumber WOPN WITH (NOLOCK) ON WOWF.WorkOrderPartNoId = WOPN.ID    
+		INNER JOIN dbo.WorkOrderManagementStructureDetails MSD WITH (NOLOCK) ON MSD.ModuleID = @ModuleID AND MSD.ReferenceID = WOPN.ID    
+		INNER JOIN DBO.WorkOrderQuote WOQ WITH (NOLOCK) ON WO.WorkOrderId = WOQ.WorkOrderId AND QuoteStatusId = @ApprovedQuoteStatusId   
+		LEFT JOIN DBO.WorkOrderQuoteDetails WQD WITH (NOLOCK) ON WOQ.WorkOrderQuoteId = WQD.WorkOrderQuoteId  
+		LEFT JOIN DBO.Customer C WITH (NOLOCK) ON C.CustomerId = WO.CustomerId  
+		LEFT JOIN dbo.EntityStructureSetup ES ON ES.EntityStructureId=MSD.EntityMSID
```
Comment (amitghediyasilverxis) [inline PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql]: Add nolock

File: PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql [hunk 11]
```diff
@@ -0,0 +1,570 @@
...
+		LEFT JOIN [dbo].TimeZone TZ WITH(NOLOCK) ON le.TimeZoneId = TZ.TimeZoneId
+	    LEFT JOIN #tmpMultipleSubWOMStocklineKit tmpWOM WITH (NOLOCK) ON tmpWOM.[SubWorkOrderId] = SWO.[SubWorkOrderId]
+		LEFT JOIN DBO.ItemMaster AS IM WITH (NOLOCK) ON tmpWOM.ItemMasterId = IM.ItemMasterId    
+		LEFT JOIN DBO.SubWorkOrderMaterialsKit AS WOM WITH (NOLOCK) ON tmpWOM.SubWorkOrderMaterialsId = WOM.SubWorkOrderMaterialsKitId
+		LEFT JOIN DBO.Stockline STK WITH (NOLOCK) ON tmpWOM.ItemMasterId = STK.ItemMasterId AND tmpWOM.ConditionId = STK.ConditionId AND STK.IsParent = 1    
+		 
+	WHERE    
+		WO.CustomerId=ISNULL(@CustomerId,WO.CustomerId)
+		AND ISNULL(WO.IsDeleted, 0) = 0  
+		--CAST((select [dbo].[ConvertUTCtoLocal] (WO.OpenDate,TZ.Description)) AS DATE) BETWEEN CAST(@Fromdate AS DATE) AND CAST(@Todate AS DATE)  AND
```
Comment (amitghediyasilverxis) [inline PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql]: Remove unnecessary code.

File: PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql [hunk 12]
```diff
@@ -0,0 +1,570 @@
...
+	WHERE    
+		WO.CustomerId=ISNULL(@CustomerId,WO.CustomerId)
+		AND ISNULL(WO.IsDeleted, 0) = 0  
+		--CAST((select [dbo].[ConvertUTCtoLocal] (WO.OpenDate,TZ.Description)) AS DATE) BETWEEN CAST(@Fromdate AS DATE) AND CAST(@Todate AS DATE)  AND  
+		 
+		AND ISNULL(WO.IsActive, 1) = 1 
+		AND  ISNULL(WO.WorkOrderStatusId, 0) != 2 -----WO Not Closed  
+		AND (ISNULL(@Stage,'') ='' OR WOS.WorkOrderStageId IN (SELECT value FROM String_split(ISNULL(@Stage,WOS.WorkOrderStageId), ',')))     
+		AND  ISNULL(WOPN.WorkOrderStatusId, 0) != 2 AND  ISNULL(WOPN.IsClosed, 0) != 1 -----MPN Not Closed  
+		--AND WO.WorkOrderId NOT IN (ISNULL(woshi.WorkOrderId,0))
```
Comment (amitghediyasilverxis) [inline PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql]: Remove unnecessary code.

**Comment by HemantPAS** [comment]
@abhishekjir @amitghediyasilverxis , Why we have missed this many places ?

File: PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql (hunk 1 above)
Comment (abhishekjir) [reply PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql]: This is done.

File: PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql (hunk 3 above)
Comment (abhishekjir) [reply PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql]: This is done

File: PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql (hunk 5 above)
Comment (abhishekjir) [reply PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql]: This is done

File: PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql (hunk 4 above)
Comment (abhishekjir) [reply PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql]: This is done

File: PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql (hunk 2 above)
Comment (abhishekjir) [reply PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql]: This is done

File: PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql (hunk 6 above)
Comment (abhishekjir) [reply PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql]: This is done

File: PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql (hunk 8 above)
Comment (abhishekjir) [reply PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql]: This is done

File: PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql (hunk 10 above)
Comment (abhishekjir) [reply PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql]: This is done

File: PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql (hunk 11 above)
Comment (abhishekjir) [reply PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql]: This is done

File: PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql (hunk 12 above)
Comment (abhishekjir) [reply PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql]: This is done

File: PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql (hunk 9 above)
Comment (abhishekjir) [reply PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql]: This is done

File: PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql (hunk 7 above)
Comment (abhishekjir) [reply PAS_Database/dbo/Stored Procedures/usprpt_GetWorkOrderAwaitingPartsReport.sql]: This is done

**Comment by amitghediyasilverxis** [comment]
Reviewed

**Comment by HemantPAS** [comment]
@abhishekjir , instead of format function use date cast function

**Pull request merged**